### PR TITLE
Add hadoop connector

### PR DESCRIPTION
**Pull request by dawidxc**

### CODE REVIEW

File: dumper/app/src/main/resources/hadoop-scripts/hive-version.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+if command -v hive > /dev/null 2>&1 ; then
+  hive --version
+fi
```
Comment (kpiotrowski) [inline dumper/app/src/main/resources/hadoop-scripts/hive-version.sh:5]: Can you please add missing new line at the end of this shell script and others?

File: dumper/app/build.gradle [hunk 2]
```diff
@@ -411,6 +412,9 @@ licenseReport {
             "commons-io:commons-io": [
                 licenseUrl: "https://raw.githubusercontent.com/apache/commons-io/master/LICENSE.txt",
             ],
+            "org.apache.commons:commons-exec": [
```
[the remark this answers is not in the document]
Comment (dawidxc) [reply dumper/app/build.gradle:415]: I plan to merge asap. If you could wait to include it in the license review, that would be great. Thank you.

File: dumper/app/src/main/resources/hadoop-scripts/hive-version.sh (hunk 1 above)
Comment (dawidxc) [reply dumper/app/src/main/resources/hadoop-scripts/hive-version.sh]: Done.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import com.google.edwmigration.dumper.plugin.ext.jdk.annotation.Description;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+@AutoService({Connector.class, MetadataConnector.class})
+@Description("Dumps metadata from the Hadoop cluster via bash commands.")
+public class HadoopMetadataConnector implements MetadataConnector {
+
+  private static final ImmutableList<String> SCRIPT_NAMES =
+      ImmutableList.of(
```
[the remark this answers is not in the document]
Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java:35]: I would like to keep the hardcoded list. In order to keep both in sync, I can add a test for the discrepancy between the list and the resource directory contents, but I would do it in a separate PR.

[the remark this answers is not in the document]
Comment (kpiotrowski) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java:35]: If you keep the hardcoded list, can we have a test to make sure all files are there?

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  public String getTargetPath() {
+    return scriptName + ".out";
+  }
+
+  private void doRun(ByteSink outputSink, ByteSink errorSink, ByteSink exitStatusSink)
+      throws IOException {
+    String scriptFilename = scriptName + ".sh";
+    URL resourceUrl = Resources.getResource("hadoop-scripts/" + scriptFilename);
+    byte[] scriptBody = Resources.toByteArray(resourceUrl);
+    File scriptDir = new File("dwh-migration-tools-tmp");
```
Comment (kpiotrowski) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java:61]: What is the reason for copying scripts into the new location?

Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java:61]: It is an extraction from jar to the local file system.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        throw e;
+      }
+    }
+  }
+
+  @Override
+  public Void run(TaskRunContext context) throws Exception {
+    OutputHandle outputHandle = createOutputHandle(context, scriptName + ".out");
+    OutputHandle errorOutputHandle = createOutputHandle(context, scriptName + ".err");
+    OutputHandle exitStatusOutputHandle = createOutputHandle(context, scriptName + ".exit-status");
```
Comment (kpiotrowski) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java:93]: What is the use case of `.exit-status` file? Do you plan to read it in assessment? Would it be possible to write error code to `.err` file? We can then assume if `.err` is empty there are no errors (status code is 0)

**Review by kpiotrowski — APPROVED** [review]

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java (hunk 4 above)
Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java:61]: I've prepared the changes in PR #425.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java:35]: I've prepared the changes in PR #425.

**Pull request merged**